doc/cookbook/time_windows.md

```diff
@@ -0,0 +1,189 @@
+# Processing Time-Windowed Data
+
+If you are analyzing data that is changing over time, chances are that you will want to perform some sort of analysis on "the last two weeks of data," "January's data," or some other moving or static time window of data.  There are a few different ways of doing these types of analyses in Pachyderm, depending on your use case.  We recommend one of the following patterns:
+
+1. [Fixed time window directory structures](#fixed-time-window-directory-structures) - for rigid, fixed time windows, such as months (Jan, Feb, etc.) or days (01-01-17, 01-02-17, etc.).
+
+2. [Rolling time window directory structures](#rolling-time-window-directory-structures) - for rolling time windows of data, such as three day windows or two week windows. 
+
+3. [Shuffled moving time windows](#shuffled-moving-time-windows) - for updating a constanly moving time-windowed data set.
+
+## Fixed time window directory structures
+
+As further discussed in [Creating Analysis Pipelines](http://docs.pachyderm.io/en/latest/fundamentals/creating_analysis_pipelines.html) and [Distributed Computing](http://docs.pachyderm.io/en/latest/fundamentals/distributed_computing.html), the basic unit of data partitioning in Pachyderm is a "datum" which is defined by a glob pattern. When analyzing data within fixed time windows (e.g., corresponding to fixed calendar times/dates), we recommend organizing your data repositories such that each of the time windows that you are going to analyze corresponds to a separate file or directory in your repository. By doing this, you will be able to:
+
+- Analyze each time window in parallel.
+- Only re-process data within a time window when that data, or a corresponding data pipeline, changes.
+
+For example, if you have monthly time windows of CSV sales data that need to be analyzed, you could create a `sales` data repository and structure it like:
+
+```
+sales
+├── January.csv
+├── February.csv
+├── March.csv
+└── April.csv
+```
+
+When you run a pipeline with a glob pattern of `/*` on `sales`, each of the months of sales data is processed in parallel (if possible).  Further, when you add a new month (e.g., `May.csv`), only that new datum, `May.csv`, will be processed.
+
+In some cases, you may also have more than one file associated with a particular time window or more than one timeframe that is relevant (e.g., both months and days).  In these cases, you would want to create a directory structure like:
+
+```
+sales
+├── January
+|   ├── online_sales.csv
+|   └── in_store_sales.csv
+├── February
+|   ├── online_sales.csv
+|   └── in_store_sales.csv
+└── March
+    ├── online_sales.csv
+    └── in_store_sales.csv
+```
+
+or, in the case when you need two time scales (months AND days):
+
+```
+sales
```

Comment:
Definitely agree on the first two examples.  Actually, I don't know what I was totally thinking there.  Sorry.  Anyway, I will modify that.

Regarding the map/reduce sort of language, I will emphasize that.  My only push back is that map/reduce language itself carries a lot of baggage, and in reality a user may have many different combinations of maps and reduces.  As such, I will add some statement in about aggregation within the time window etc, but will likely avoid too much distraction with map/reduce.